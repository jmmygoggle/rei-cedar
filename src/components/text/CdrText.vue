<template>
  <component
    :is="tag"
    :class="[$style[baseClass], modifierClass]"
  >
    <slot />
  </component>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue';

import { buildClass } from '../../utils/buildClass';

export default defineComponent({
  name: 'CdrText',
  props: {
    tag: {
      type: String,
      default: 'p',
    },
    modifier: {
      type: String,
      required: false,
    },
  },
  setup(props) {
    const baseClass = 'cdr-text';
    const modifierClass = computed(() => buildClass(baseClass, props.modifier));

    return {
      baseClass,
      modifierClass,

    };
  },
});
</script>

<style lang="scss" module src="./styles/CdrText.scss">
</style>
