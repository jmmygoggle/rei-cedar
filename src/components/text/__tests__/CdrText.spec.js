import { mount } from '../../../../test/vue-jest-style-workaround.js';
import { h } from 'vue';
import CdrText from 'componentdir/text/CdrText';

describe('CdrText', () => {
  test('renders correctly', () => {
    const wrapper = mount(CdrText, {
      slots: {
        default: h('paragraph')
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders as h1 correctly', () => {
    const wrapper = mount(CdrText, {
      propsData: {
        tag: 'h1',
      },
      slots: {
        default: 'header'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
