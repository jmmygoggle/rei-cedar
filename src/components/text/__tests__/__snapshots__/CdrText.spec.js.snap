// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrText renders as h1 correctly 1`] = `
<h1
  class="cdr-text"
>
  
  header
  
</h1>
`;

exports[`CdrText renders correctly 1`] = `
<p
  class="cdr-text"
>
  
  paragraph
  
</p>
`;
